internal/config/task_config.go

```diff
@@ -0,0 +1,88 @@
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+
+	"gopkg.in/yaml.v3"
+)
+
+const defaultConfigPath = "config/tasks.yaml"
+
+type TaskConfig struct {
+	Dependencies map[string]Dependency
+	Tasks        []Task `yaml:"tasks"`
+	ConfigPath   string
+}
+
+type Dependency struct {
+	Host string `yaml:"host"`
+	Port int    `yaml:"port"`
+	TLS  bool   `yaml:"tls"`
+}
+
+type Task struct {
+	Name    string `yaml:"name"`
+	Command string `yaml:"command"`
+}
+
+func (t *TaskConfig) MustLoad() {
```

Comment:
I wanted the config loading to fail fast if it did fail - I'll think about this a bit more after I work through the other items. This could be another issue with my thought process